presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergEqualityDeleteAsJoin.java

```diff
@@ -0,0 +1,467 @@
+                    outputsBuilder.build(),
+                    assignmentsBuilder.build(),
+                    node.getTableConstraints(),
+                    node.getCurrentConstraint(),
+                    node.getEnforcedConstraint());
+        }
+
+        /**
+         * Calculate required fields that the user didn't include in his select statement and add assignments for them to add to the table scan
+         */
+        private ImmutableMap<VariableReferenceExpression, ColumnHandle> createAssignmentsForUnselectedFields(TableScanNode node,
+                ImmutableMap<Set<Integer>, DeleteSetInfo> deleteInfos,
+                Table icebergTable)
+        {
+            Set<Integer> selectedFields = node.getAssignments().values().stream().map(f -> ((IcebergColumnHandle) f).getId()).collect(Collectors.toSet());
+            Set<Integer> unselectedFields = Sets.difference(deleteInfos.keySet().stream().reduce(Sets::union).orElseGet(Collections::emptySet), selectedFields);
+            ImmutableMap.Builder<VariableReferenceExpression, ColumnHandle> unselectedAssignmentsBuilder = ImmutableMap.builder();
+            ImmutableMap<Integer, PartitionFieldInfo> partitionFields = deleteInfos.values().stream()
+                    .map(DeleteSetInfo::getPartitionFields)
+                    .reduce((a, b) -> new ImmutableMap.Builder<Integer, PartitionFieldInfo>().putAll(a).putAll(b).build())
+                    .orElseGet(ImmutableMap::of);
+            unselectedFields
+                    .forEach(fieldId -> {
+                        if (partitionFields.containsKey(fieldId)) {
+                            PartitionFieldInfo partitionFieldInfo = partitionFields.get(fieldId);
+                            PartitionField partitionField = partitionFieldInfo.getPartitionField();
+                            Types.NestedField sourceField = icebergTable.schema().findField(partitionField.sourceId());
+                            if (!partitionField.transform().isIdentity()) {
+                                if (!partitionField.transform().isIdentity()) {
+                                    Type partitionFieldType = partitionField.transform().getResultType(sourceField.type());
+                                    VariableReferenceExpression variableReference = variableAllocator.newVariable(partitionField.name(), toPrestoType(partitionFieldType, typeManager));
+                                    IcebergColumnHandle columnHandle = new IcebergColumnHandle(
+                                            ColumnIdentity.createColumnIdentity(partitionField.name(), partitionField.fieldId(), partitionFieldType),
+                                            toPrestoType(partitionFieldType, typeManager),
+                                            Optional.empty(),
+                                            PARTITION_KEY);
+                                    unselectedAssignmentsBuilder.put(variableReference, columnHandle);
```

Comment:
Are non-identity partition columns necessary to add to  `unselectedAssignmentsBuilder`? As I  understand, they would not be used in join equality clause with  equal delete table scan.

Reply:
They are added to the join clause. Only the metadata column "$data_sequence_number" isn't. (that column is used as a filter instead).

The test `testEqualityDeletesWithHiddenPartitions` covers this case.

Reply:
Oh, I overlooked the return statement after identity partition column branch in `DeleteSetInfo.allFields(Schema schema)`.